src/pages/newCommunity/NewCommunity.jsx

```diff
@@ -0,0 +1,98 @@
+import React from "react";
+import "./newCommunity.css";
+import Grid from "@mui/material/Grid";
+import TextField from "@mui/material/TextField";
+import Button from "@mui/material/Button";
+import AddPhotoAlternateIcon from "@mui/icons-material/AddPhotoAlternate";
+import { useState, useEffect } from "react";
+const NewCommunity = () => {
+  const [selectedFile, setSelectedFile] = useState();
+  const [preview, setPreview] = useState();
+
+  // create a preview as a side effect, whenever selected file is changed
+  useEffect(() => {
+    if (!selectedFile) {
+      setPreview(undefined);
+      return;
+    }
+
+    const objectUrl = URL.createObjectURL(selectedFile);
+    setPreview(objectUrl);
+
+    // free memory when ever this component is unmounted
+    return () => URL.revokeObjectURL(objectUrl);
+  }, [selectedFile]);
+
+  const onSelectFile = (e) => {
+    if (!e.target.files || e.target.files.length === 0) {
+      setSelectedFile(undefined);
+      return;
+    }
+
+    setSelectedFile(e.target.files[0]);
+  };
```

Comment:
Following a discussion with @Xiaoqi510 a change has been requested to convert the image to a base64 string before storing it into a variable, which requires additional code in this area. This is to allow for the image to be stored efficiently in the backend.

Reply:
okay, fixed it. thank you for your advice